wordpress-mobile/WordPress-iOS
# Adding Categories to Prepublishing bottom sheet

This PR adds categories row to prepublishing bottom sheet.

Previously the Prepublishing navigation controller determined a fixed height `.contentHeight(Constants.height)` in the bottomsheet. That worked well for screens that mostly maintain their height (like PostTagPickerViewController) but since the categories table view has a dynamic height, we have to update the container view whenever the height of the presented changes. 
Because of that I updated the navigation collapsedHeight to be calculated (.intrinsicHeight) based on the content of the presented and thats why I added the preferredContentSize override in the navigation's view controllers . 

To test:
1. Go to the editor 
2. Open a draft post 
3. Click More (3 dots) 
4. Tap Publish 
5. See categories row in bottom sheet
6. Tap Categories 
7. See Post Categories screen
8. Tap + 
9. See add categories screen 
10. Try adding, removing, checking, unchecking categories
11. See `editor_post_category_changed <via: prepublishing_nudges>` event fired
11. Repeat 2-10 on iPad
12. tap on Tags and other rows 

![prepublishing_categories](https://user-images.githubusercontent.com/1335657/98453045-ce083a00-2109-11eb-9e75-fae23158cd71.gif)



PR submission checklist:
